tests/test_runner.py

```diff
@@ -123,11 +120,52 @@ def test_runner(key):
             break
     else:
         assert False, 'Did not successfully run kernel'
+
+    return valid, outputs
+
+
+@pytest.mark.parametrize("key", find_test_keys())
+def test_runner(key):
+    kernel_manager = provider.make_manager(key)
+    if kernel_manager.kernel_spec.argv[:3] == RUNNER_COMMAND:
+        env_path = kernel_manager.kernel_spec.argv[4]
+    else:
+        env_path = sys.prefix
+    env_path = os.path.normcase(os.path.normpath(env_path))
+
+    valid, outputs = call_kernel(kernel_manager)
+
     assert valid and len(outputs) >= 2
     for o in outputs[-2:]:
         assert os.path.normcase(os.path.normpath(o)) == env_path, (o, env_path)
 
 
+@pytest.mark.parametrize("jupyter_kernel", find_test_keys(), indirect=True)
+def test_jupyter_kernelspecs_runner(tmp_path, jupyter_kernel):
+    if sys.platform.startswith("linux") and jupyter_kernel.kernel_name == "conda-env-t_st_env2-py":
```

Comment:
I haven't seen this issue before. Do you have details you can share? I would have thought Linux is the easy one to get right here. Windows is the one that always gives me pain.

Reply:
Yeah, I was very surprise too.

The initial error is: 

```
>       names = os.listdir(src)
E       FileNotFoundError: [Errno 2] No such file or directory: '/home/runner/work/nb_conda_kernels/conda/envs/tæst_env2/share/jupyter/kernels/python3'
```

in `shutil.copytree`:

```
../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.6/site-packages/jupyter_client/kernelspec.py:345: in install_kernel_spec
    shutil.copytree(source_dir, destination)
```

Reference: https://github.com/Anaconda-Platform/nb_conda_kernels/runs/1137417932

Googling the problem, there are some reports of the problem: https://vstinner.github.io/python30-listdir-undecodable-filenames.html.

I tried passing bytes. But then this is the jupyter code that is complaining: https://github.com/Anaconda-Platform/nb_conda_kernels/runs/1140670111

Reply:
OK, we can merge without fixing this but I want to create an issue to address it because it does represent an apparent regression.